lib/prepare.js

```diff
@@ -769,12 +776,32 @@ function updateIconResourceForAdaptive (preparedIcons, resourceMap, platformReso
             resourceMap[targetPathForeground] = android_icons[density].foreground;
         }
 
+        if (path.extname(path.basename(monochrome)) === '.xml') {
+            // Vector Use Case
+            targetPathMonochrome = getAdaptiveImageResourcePath(platformResourcesDir, 'mipmap', density, 'ic_launcher_monochrome.xml', path.basename(android_icons[density].monochrome));
+            resourceMap[targetPathMonochrome] = android_icons[density].monochrome;
+        } else if (path.extname(path.basename(monochrome)) === '.png') {
+            // Images Use Case
+            targetPathMonochrome = getAdaptiveImageResourcePath(platformResourcesDir, 'mipmap', density, 'ic_launcher_monochrome.png', path.basename(android_icons[density].monochrome));
+            resourceMap[targetPathMonochrome] = android_icons[density].monochrome;
+        }
```

Comment:
I think this if/elseif block needs to be contained in a `if (monochrome)` block, given my review points above.